### ISSUE TITLE
Generic default value generates newDeref error

### ISSUE THREAD

**Issue by Bechma** [go-critic/go-critic#1286]
### Description of the problem

As stated in this post of stackoverflow https://stackoverflow.com/questions/70585852/return-default-value-for-generic-type/70589302#70589302

```go
func Zero[T any]() T {
    return *new(T)
}
```

```go
func IsZero[T comparable](v T) bool {
    return v == *new(T)
}
```

I tried to apply the `new` idiom approach to generate default values for generic functions but I got an error from `golangci-lint`
```
runner/v2/runner.go:101:38: newDeref: replace `*new(T)` with `T(nil)` (gocritic)
                        r.results <- &RunResult[T]{Value: *new(T), Err: err, Idx: idx}
```

T(nil) is not valid for the compiler, so the replace hint is not valid. Can we make an exception for generics?

This is the output of gocritic:
$ go install -v github.com/go-critic/go-critic/cmd/gocritic@latest
...
$ gocritic version
gocritic version: v0.0.0-SNAPSHOT
$ gocritic check ./...
./runner/v2/runner.go:101:38: newDeref: replace `*new(T)` with `T(nil)`
```

**Comment by quasilyte** [comment]
See https://github.com/go-critic/go-critic/issues/1284
https://github.com/go-critic/go-critic/issues/1272

**Comment by Bechma** [comment]
Oh, I didn't reach those issues, I'm sorry, closing.
Strange that even I compile from sources I still get that error. I'll wait for the new release

**Comment by quasilyte** [comment]
Hmm, there should be no warning if you're running gocritic from master.
Let's keep this issue open until we learn why it happens.
The fix should be in the main branch already.

**Comment by cristaloleg** [comment]
Is this still an issue?

**Comment by Bechma** [comment]
@cristaloleg I just install again gocritic from sources and it's not producing any warning, so the problem was fixed correctly.
